samcli/lib/config/samconfig.py

```diff
@@ -39,11 +40,18 @@ def __init__(self, config_dir, filename=None):
             Optional. Name of the configuration file. It is recommended to stick with default so in the future we
             could automatically support auto-resolving multiple config files within same directory.
         """
+        self.document = {}
         self.filepath = Path(config_dir, filename or DEFAULT_CONFIG_FILE_NAME)
+        self.file_manager = self.FILE_MANAGER_MAPPER.get(self.filepath.suffix[1:], None)  # default to TOML
```

Comment:
I agree that if there are multiple configs and one isn't specified we should default to TOML. But if a customer only has one configuration file that isn't the default, should we just use that one?

Reply:
I think that comment is outdated from previously, where the `file_manager` would default to `TomlFileManager` if the given config file isn't supported, which I resolved in a previous commit. If the customer has a config file, it'll be set and passed into the constructor via `cli_config_file.py::TomlProvider::__call__`, which passes in the `config_path` which is defined if there's an existing config. Only if that isn't defined will it use the default name

Reply:
As discussed in our 1:1 earlier this morning, that logic will be put into `SamConfig` after all the other `FileManager` subclasses are created. Expected behaviour for that is that it will first check if a config file is provided and exists, then if a TOML exists, then YAML, then JSON, and if none of those exist, default to TOML.